### PR TITLE
This fixes a way to update plugins in jenkins when required

### PR DESCRIPTION
**Pull request by rupalibehera**
copy plugins when required, so that we could update any required plugins. It will fix https://github.com/openshiftio/openshift.io/issues/3393

### CODE REVIEW

**Comment by rupalibehera** [comment]
Also the version.txt added in plugins direcoty https://github.com/fabric8io/openshift-jenkins-s2i-config/pull/164

**Comment by rupalibehera** [comment]
> Can you update README as well? https://github.com/fabric8-jenkins/jenkins-openshift-base#plugins with this update.txt thing? I know that it is not something of this repo but I prefer to have this thing documented everywhere.

@lordofthejars 
I updated the README

File: 2/contrib/s2i/run [hunk 1]
```diff
@@ -226,12 +226,31 @@ if [[ -n "${INSTALL_PLUGINS:-}" ]]; then
...
-    echo "Copying $(ls /opt/openshift/plugins | wc -l) Jenkins plugins to ${JENKINS_HOME} ..."
-    cp -frL /opt/openshift/plugins/* ${JENKINS_HOME}/plugins/
+      copy_plugins
+  else
+    if [ ! -f ${JENKINS_HOME}/plugins/version.txt ]; then
+      copy_plugins
+    else
+      previous_version=$(<${JENKINS_HOME}/plugins/version.txt)
+      new_version=$(</opt/openshift/plugins/version.txt)
+      if [ $new_version -gt $previous_version ]; then
```
Comment (maxandersen) [inline 2/contrib/s2i/run]: to make it easy to debug print out the previous new versions found before this line is evaluated.

File: 2/contrib/s2i/run [hunk 2]
```diff
@@ -226,12 +226,31 @@ if [[ -n "${INSTALL_PLUGINS:-}" ]]; then
   /usr/local/bin/install-plugins.sh "${plugins_file}"
 fi
 
+copy_plugins() {
+    echo "Copying $(ls /opt/openshift/plugins | wc -l) Jenkins plugins to ${JENKINS_HOME} ..."
+    cp -frL /opt/openshift/plugins/* ${JENKINS_HOME}/plugins/
```
Comment (maxandersen) [inline 2/contrib/s2i/run]: does this copy *everything* and will trigger unzip/reload of all plugins or just the changed ones ?

just to understand how big impact we make here.

Comment (rupalibehera) [reply 2/contrib/s2i/run]: it will copy everything

**Pull request merged**